### DISCUSSION TITLE
Flip Index Error (COMMIT ERROR)

### DISCUSSION THREAD

**Discussion by ogarden** [ultralytics/ultralytics#2865] [General]
Hello,

I have been training YOLOv8 models recently for a personal project, and suddenly today my training framework stopped working. I was getting an error saying: "data.yaml flip_idx={flip_idx} must be zero-index (start from 0)". This error didn't make sense to me, so I checked the GitHub, and sure enough, a commit was made 2 weeks ago that added this error check. However, I believe this commit was a mistake. After all, what is this code trying to do?:

```
elif flip_idx[0] != 0:
                raise ValueError(f'data.yaml flip_idx={flip_idx} must be zero-index (start from 0)')
```

In my specific case, the first keypoint needs to be swapped because when the frames are horizontally flipped, the 0th keypoint is swapped with the 2nd keypoint. However, this error check requires that the first swapped keypoint needs to still be 0? Was this intended? Or were you guys trying to accomplish something else? I had to edit the source code on my local machine to get my model to train. After all, I needed the first index of the flip index to not be 0.

Best,
Oliver Traben

**Comment by glenn-jocher** [comment]
@ogarden thanks for reporting this! Yes you are right this check is causing problems. I've removed it in Ultralytics 8.0.110 scheduled for publishing later today. Please let us know if there are any other issues or areas we could improve!

Also in the future please submit a PR with a fix directly if you find problems, thanks!

<img width="1057" alt="Screenshot 2023-05-28 at 18 04 48" src="https://github.com/ultralytics/ultralytics/assets/26833433/15d34679-9055-4bd0-9b6d-6418fb1317c9">
